src/handler.rs

```diff
@@ -633,6 +662,85 @@ fn calculate_fee(fee_percent: f64, principle: U256) -> U256 {
     fee_u256
 }
 
+pub fn handle_substrate<'a>(
+    ctx: RelayerContext,
+    cmd: SubstrateCommand,
+) -> BoxStream<'a, CommandResponse> {
+    match cmd {
+        SubstrateCommand::MixerRelayTx(cmd) => {
+            handle_substrate_mixer_relay_tx(ctx, cmd)
+        }
+    }
+}
+
+fn handle_substrate_mixer_relay_tx<'a>(
+    ctx: RelayerContext,
+    cmd: MixerRelayTransaction,
+) -> BoxStream<'a, CommandResponse> {
+    use webb::substrate::protocol_substrate_runtime::DefaultConfig;
+    use webb::substrate::protocol_substrate_runtime::RuntimeApi;
+    use CommandResponse::*;
+
+    let s = stream! {
+        let requested_chain = cmd.chain.to_lowercase();
+        let chain: SubstrateConfig = match ctx.config.substrate.get(&requested_chain) {
+            Some(v) => v,
+            None => {
+                yield Network(NetworkStatus::UnsupportedChain);
+                return;
+            }
+        };
+
+        let api = subxt::ClientBuilder::new()
+            .set_url(chain.http_endpoint)
+            .build()
+            .await?
+            .to_runtime_api::<RuntimeApi<DefaultConfig>>();
```

Comment:
Try this one is better, as in the future we maybe cache all these clients instead of open a new connection everytime: 
```suggestion
        let client = ctx.substrate_provider::<DefaultConfig>(&requested_chain).await?;
        let api = client.to_runtime_api::<RuntimeApi<DefaultConfig>>();
```